### PR TITLE
Update node add hooks

### PR DESCRIPTION
**Pull request by tejasv02**
# Contribution description
> Please add a brief description of the contents of the pull request

# Pull request checklist
- [x] Contributed code respects the [editorconfig rules](.editorconfig)
- [x] Contributed code passes the [eslint rules](.eslintrc.yaml)
- [x] Contributed code passes the unit tests
- [x] Added rules are described in the [readme file](README.md)
- [x] [The build](https://travis-ci.org/webdriverio/cucumber-boilerplate) of the PR is passing

### CODE REVIEW

**Comment by christian-bromann** [comment]
@tejasv02 can you please elaborate on the intention of this change?

**Comment by tejasv02** [comment]
version bump, plus segregate hooks.
As hooks are part of cucumber and will have a different implementation based on the platform used like web/mobile. Segregating them will help up load it on a need basis.

File: wdio.conf.js [hunk 1]
```diff
@@ -159,6 +160,7 @@ exports.config = {
             './src/steps/given.js',
             './src/steps/then.js',
             './src/steps/when.js',
+            './src/support/hooks.js',
```
Comment (christian-bromann) [inline wdio.conf.js:163]: Why do you require hooks in here?

Comment (tejasv02) [reply wdio.conf.js:163]: Sorry unintended change

Comment (tejasv02) [reply wdio.conf.js:163]: I have pushed change onto same branch but it doesn't reflect on this merge request, Hence I have created duplicate MR

**Comment by tejasv02** [comment]
travis builds are still failing

**Comment by christian-bromann** [comment]
@tejasv02 this is due to an invalid Chromedriver version, can you check which Chrome version is being used by Travis and update the Chromedriver dependency accordingly?

**Comment by tejasv02** [comment]
I can try :)

File: wdio.conf.js [hunk 2]
```diff
@@ -129,7 +130,11 @@ exports.config = {
     // The only one supported by default is 'dot'
     // see also: https://webdriver.io/docs/dot-reporter.html
     reporters: ['spec'],
-    //
+    services: [
+        ['chromedriver', {
```
Comment (christian-bromann) [inline wdio.conf.js:134]: this should go into the build config file where the chromedriver service is already defined: https://github.com/webdriverio/cucumber-boilerplate/blob/master/wdio.BUILD.conf.js#L23-L26 .. I would recommend to remove this change.

File: wdio.conf.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 const path = require('path');
+const { hooks } = require('./src/support/hooks');
 
-exports.config = {
+exports.config = Object.assign({}, {
```
Comment (christian-bromann) [inline wdio.conf.js:4]: How about:
```js
exports.config = {
  // original config
  ...hooks
}
```

File: .travis.yml [hunk 4]
```diff
@@ -28,6 +28,10 @@ branches:
     - master
     - /^greenkeeper/.*$/
 
+before_script:
+    "google-chrome --product-version"
+    "echo google-chrome --product-version"
```
Comment (christian-bromann) [inline .travis.yml:33]: are these necessary?

Comment (tejasv02) [reply .travis.yml:33]: sorry I was checking which version of chrome was present

File: wdio.conf.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 const path = require('path');
+const { hooks } = require('./src/support/hooks');
 
-exports.config = {
-    //
+exports.config = { //
```
Comment (christian-bromann) [inline wdio.conf.js:4]:
```suggestion
exports.config = {
    //
```

File: src/support/hooks.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
+//
+// =====
+// Hooks
+// =====
+// WebdriverIO provides a several hooks you can use to interfere the test process in order to
+// enhance it and build services around it. You can either apply a single function to it or
+// an array of methods. If one of them returns with a promise,
+// WebdriverIO will wait until that promise is resolved to continue.
+//
+
```
Comment (christian-bromann) [inline src/support/hooks.js:10]:
```suggestion
```

**Pull request merged**